src/context-selection/attribute-option-combo-selector-bar-item/use-category-combination.js

```diff
@@ -46,47 +35,60 @@ const isOptionWithinPeriod = ({
 export default function useCategoryCombination() {
     const [periodId] = usePeriodId()
     const [dataSetId] = useDataSetId()
-    const queryKey = [QUERY_CATEGORY_COMBINATION, { id: dataSetId }]
+    const params = {
+        fields: [
+            'categoryCombo[isDefault,displayName, categories[id, displayName, categoryOptions[id, displayName]]]',
+        ],
+    }
+    const queryKey = [
+        'dataSets',
+        {
+            id: dataSetId,
+            params,
+        },
+    ]
     const {
         isIdle,
         isLoading: loading,
         error,
         data,
     } = useQuery(queryKey, {
-        enabled: !!dataSetId,
-    })
+        enabled: !!dataSetId && !!periodId,
+        select: (data) => {
+            if (!data || !periodId) {
+                return data?.categoryCombo
+            }
 
-    const categoryCombo = useMemo(() => {
-        if (!data || !periodId) {
-            return data?.dataSet.categoryCombo
-        }
```

Comment:
I've removed this condition as this _should_ never happen now we've moved to using the selector. Because the query is disabled if there's no periodId, and if there's no data yet the selector won't be called.

Reply:
:+1: